crates/metal-codegen-llvm/src/stmt/function_definition.rs

```diff
@@ -11,15 +11,20 @@ use llvm_sys::{
 use metal_mir::stmt::functiondef::FunctionDefinition;
 
 use super::{CodeGenType, CodeGenValue};
-use crate::get_linkage_from_vis;
+use crate::{core::get_module_full_name, get_linkage_from_vis};
 
-impl CodeGenValue for FunctionDefinition {
+impl CodeGenValue for FunctionDefinition<'_> {
     fn llvm_value(
         &self,
         llvm: &mut crate::LLVMRefs,
         module: &metal_mir::parcel::Module,
     ) -> LLVMValueRef {
-        let fun_name = self.signature.name.as_str();
+        let fun_name = if self.signature.name != "main" {
+            let module_name = get_module_full_name(module);
+            String::new() + module_name.as_str() + "." + self.signature.name.as_str()
```

Comment:
Can't `module_name` be used instead of `String::new()`?

Reply:
Fixed